Cargo.toml

```diff
@@ -51,3 +51,8 @@ tracing = { version = ">=0.1.40", default-features = false }
 tracing-core = { version = ">=0.1.33", default-features = false } 
 tracing-subscriber = { version = "0.3", default-features = false }
 url = { version = "2.5", default-features = false }
+
+[patch.crates-io]
+opentelemetry = { path = "opentelemetry" }
+opentelemetry_sdk = { path = "opentelemetry-sdk" }
+opentelemetry-stdout = { path = "opentelemetry-stdout" }
```

Comment:
Looks like the build failure

```
     Checking opentelemetry-prometheus v0.27.0 (/home/runner/work/opentelemetry-rust/opentelemetry-rust/opentelemetry-prometheus)
error[E0624]: associated function `empty` is private
   --> opentelemetry-prometheus/src/lib.rs:287:33
    |
287 |             resource: Resource::empty(),
    |                                 ^^^^^ private associated function
    |
   ::: /home/runner/work/opentelemetry-rust/opentelemetry-rust/opentelemetry-sdk/src/resource/mod.rs:82:5
    |
82  |     pub(crate) fn empty() -> Self {
    |     ----------------------------- private associated function defined here
```

is due to opentelemetry-prometheus not using path dependencies for opentelemetry/opentelemetry_sdk and being incompatible with the workspace version of these crates (e.g. https://github.com/open-telemetry/opentelemetry-rust/pull/2322 seems to be the cause of this incompatibility), so this PRs dependency override causes it to actually use the workspace versions which it can't compatible with.

Should opentelemetry-prometheus be using the workspace versions of these crates?  If it isn't desired for all crates in this workspace to be tested against the workspace version, then should opentelemetry-appender-tracing also get tested against the latest release on crates.io?

Reply:
`opentelemetry-prometheus` is not in active development, and it won't _always_ work with the local workspace version. Can we have this to ignore the dependency override?

Reply:
> Can we have this to ignore the dependency override?

I don't see a way to scope the patch so it doesn't apply to a crate in the workspace and cargo requires the patch be applied at the root of the workspace (i.e. not just to a specific crate in the workspace).

One idea would be to remove it as a member of the workspace, but continue to have CI test it (e.g. add another `cargo test` command to scripts/test.sh).  A glob is used to specify the workspace members, but `exclude = ["opentelemetry-*"]` can be added to the `[workspace]` section.  Would this be an acceptable solution to this problem?  Or does any have any better ideas?

Reply:
> One idea would be to remove it as a member of the workspace, but continue to have CI test it (

this looks fine to me, specifically since the fate of Prometheus crate is still under discussion #2451. @cijothomas @TommyCpp  do you see any concerns, or better option here.